addons/mail/static/src/models/activity/activity.js

```diff
@@ -150,7 +150,12 @@ function factory(dependencies) {
             };
             this.env.bus.trigger('do-action', {
                 action,
-                options: { on_close: () => this.fetchAndUpdate() },
+                options: {
+                    on_close: () => {
+                        this.fetchAndUpdate();
+                        typeof afterUpdateCallback === 'function' && afterUpdateCallback();
```

Comment:
just check existence, something else than function shouldn't be given

and avoid hiding instructions with side effect after boolean operations, prefer explicit condition